share/getters/shrex.go

```diff
@@ -135,6 +194,7 @@ func (sg *ShrexGetter) GetSharesByNamespace(
 				"hash", root.String(),
 				"err", getErr,
 				"finished (s)", time.Since(start))
+			sg.metrics.recordNDAttempt(ctx, attempt, false)
```

Comment:
the attempt count is only recorded once, before returning (since that was the amount of attempts necessary before either giving up or returning successfully)